google-cloud-bigquery/lib/google/cloud/bigquery/service.rb

```diff
@@ -340,8 +352,10 @@ def self.table_ref_from_s str, default_table_ref
             dataset_id: m["dts"],
             table_id:   m["tbl"]
           }.delete_if { |_, v| v.nil? }
-          new_table_ref_hash = default_table_ref.to_h.merge str_table_ref_hash
-          Google::Apis::BigqueryV2::TableReference.new new_table_ref_hash
+          if default_ref
+            str_table_ref_hash = default_ref.to_h.merge str_table_ref_hash
+          end
+          Google::Apis::BigqueryV2::TableReference.new str_table_ref_hash
```

Comment:
Should an `ArgumentError` be raised here if `str_table_ref_hash` does not have 3 keys? For instance, if the `str` `"my-table-id"` was passed along with a `ProjectReference` object, there would be no `dataset_id` in the hash. In that case, should `ArgumentError` be raised? Or should we allow the API to raise an appropriate error?

AFAICT this was never an issue in `Dataset::Access` because we always had a `DatasetReference`.

Reply:
The "appropriate error" that you get when the `TableReference` is missing a field is not very helpful:

```sh
Google::Cloud::InvalidArgumentError: required: Required parameter is missing
```

So I think raising `ArgumentError` here is a good idea.